File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        maps = {}

        for i in range(len(nums)):
            rem = target - nums[i]

            if rem not in maps.keys():
                maps[nums[i]] = i
            
            else:
                return [maps[rem], i]
        

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const myMap = new Map()
    for(let i = 0; i< nums.length; i++){
        let rem = target - nums[i]
        if(myMap.has(rem)) return [myMap.get(rem), i]
        else {
            myMap.set(nums[i], i)
        }
    }
};